### PR TITLE
bitlbee-plugins: use runCommandLocal instead of mkDerivation

### PR DESCRIPTION
**Pull request by Stunkymonkey**
###### Motivation for this change
#28910

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Stunkymonkey** [comment]
maybe one of @jpotier, @toonn, @lassulus, @jb55 can have a look at this, because they are maintainer of a package using this script.

**Comment by siraben** [comment]
Result of `nixpkgs-review pr 134899` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)

**Review by siraben — APPROVED** [review]

**Comment by toonn** [comment]
I'll try to test and report on this tomorrow.

**Comment by toonn** [comment]
Looks like the `bitlbee-facebook` plugin stopped working with this change.
I only used an overlay on my old working setup though so I can't say with absolute certainty this is broken. Warrants more testing at least though.

```
self: super: {
  bitlbee-plugins = (import (super.fetchFromGitHub {  #  owner = "Stunkymonkey";  #  repo = "nixpkgs";
    rev = "bitlbee-plugins-phases";
    sha256 = "10b68mad8qmq3hq8y3c4j3nvh0q2jw8bnqz4w0lhdhxbg48lirpf";
  }) {}).pkgs.bitlbee-plugins;

  bitlbee-facebook = super.bitlbee-facebook.overrideAttrs (oas: {
      src = super.fetchFromGitHub {        owner = "bitlbee";
        repo = "bitlbee-facebook";
        rev = "49ea312d98b0578b9b2c1ff759e2cfa820a41f4d";
        sha256 = "0zg1p9pyfsdbfqac2qmyzcr6zjibwdn2907qgc808gljfx8bfnmk";
      };
    });

  purple-matrix = super.callPackage
    ( { lib, stdenv, fetchFromGitHub, pkg-config, pidgin, json-glib
      , glib, http-parser, sqlite, olm, libgcrypt
      }:
      stdenv.mkDerivation rec {
        pname = "purple-matrix-unstable";
        version = "2019-06-06";
        src = fetchFromGitHub {
          owner = "matrix-org";
          repo = "purple-matrix";          rev = "4494ba22b479917f0b1f96a3019792d3d75bcff1";
          sha256 = "1gjm0z4wa5vi9x1xk43rany5pffrwg958n180ahdj9a7sa8a4hpm";
        };
        NIX_CFLAGS_COMPILE = builtins.toString [
          # glib-2.62 deprecations
          "-DGLIB_DISABLE_DEPRECATION_WARNINGS"
          # override "-O0 -Werror" set by build system          "-O3" "-Wno-error"        ];
        nativeBuildInputs = [ pkg-config ];
        buildInputs = [ pidgin json-glib glib http-parser sqlite olm libgcrypt ];

        makeFlags = [
          "PLUGIN_DIR_PURPLE=${placeholder "out"}/lib/purple-2"
          "DATA_ROOT_DIR_PURPLE=${placeholder "out"}/share"        ];

        buildFlags = [ "CC=${stdenv.cc.targetPrefix}cc" ]; # fix build on darwin
        meta = with lib; {
          homepage = "https://github.com/matrix-org/purple-matrix";
          description = "Matrix support for Pidgin / libpurple";
          license = licenses.gpl2;          platforms = platforms.unix;
          maintainers = with maintainers; [ symphorien ];
        };
      }    ) {};
}
```

**Comment by Stunkymonkey** [comment]
does `bitlbee-facebook` work without my change? first we have to make sure, that facebook is not the problem.

**Comment by toonn** [comment]
The version I was using before does and still does after a rollback. This is on a pentium with 450MB of RAM so doing more builds to test is not really an option for me unfortunately : s

File: pkgs/applications/networking/instant-messengers/bitlbee/plugins.nix [hunk 1]
```diff
@@ -1,20 +1,15 @@
...
 
 with lib;
 
-plugins:
-
-stdenv.mkDerivation {
-  inherit bitlbee plugins;
-  name = "bitlbee-plugins";
+plugins: runCommandLocal "bitlbee-plugins"
+{
```
Comment (figsoda) [inline pkgs/applications/networking/instant-messengers/bitlbee/plugins.nix:6]:
```suggestion
{
  inherit plugins;
```

**Review by figsoda — CHANGES_REQUESTED** [review]
maybe this will fix the issue?

File: pkgs/applications/networking/instant-messengers/bitlbee/plugins.nix (hunk 1 above)
Comment (figsoda) [reply pkgs/applications/networking/instant-messengers/bitlbee/plugins.nix:6]: also, `{` shouldn't have a newline in the front
https://nixos.org/manual/nixpkgs/unstable/#sec-syntax

Comment (Stunkymonkey) [reply pkgs/applications/networking/instant-messengers/bitlbee/plugins.nix]: good idea. should be now as expected.

**Review by figsoda — APPROVED** [review]

**Comment by figsoda** [comment]
@toonn does this fix the issue?

**Comment by toonn** [comment]
I will try to build it, bear with me though this is an old machine so it runs out of memory and takes ages.

**Pull request merged**